teps/0001-tekton-enhancement-proposal-process.md

```diff
@@ -0,0 +1,464 @@
+
+- [Tekton Pipeline Resource Extensibility](https://docs.google.com/document/d/1rcMG1cIFhhixMSmrT734MBxvH3Ghre9-4S2wbzodPiU/edit#)
+- [Why Aren't PipelineResources in Beta ?](https://github.com/tektoncd/pipeline/blob/master/docs/resources.md#why-arent-pipelineresources-in-beta)
+- [Pipeline Reosurces Redesign](https://github.com/tektoncd/pipeline/issues/1673)
+
+1. A TEP is proposed to add extensibility to PipelineResources. This
+   is based on [Tekton Pipeline Resource
+   Extensibility](https://docs.google.com/document/d/1rcMG1cIFhhixMSmrT734MBxvH3Ghre9-4S2wbzodPiU/edit#)
+   that has been discussed among the community. The initial idea is to
+   make the PipelineResource extensible and not be limited to built-in
+   options.
+2. The TEP is accepted as `proposed`, which means:
+   - We acknowledge this is important for the project, and needs to be
+     worked on
+   - It needs some work and discussion based on the initial proposal
+3. The TEP gets discussed at length during a special Working
+   Group. After multiple iteration, it becomes clear that:
+
+   - The current PipelineResource design has some limits and problems
+   - The current proposed TEP is way too complicated
+   - A life is possible without using PipelineResource, some
+     experimentation needs to be done around this
+
+   The next action are:
+   - Mark this TEP as withdrawn, we acknoledge it is not the way to
+     go. When marking this as `withdrawn`, add the reason why.
+   - Conduct experiment on not using PipelineResource
+   - Act that the `PipelineResource` needs a full re-design (and thus
+     removing it from the beta API for now)
+4. From the conducted experiment on "a life without PipelineResource",
+   two concept are being discussed:
+   - Workspace : to share data between tasks
+   - Results : to share results between tasks
+
+   A TEP for each is created, approved and implemented.
+5. Design discussion and docs are being created to re-design
+   PipelineResources using the above new concept. It gets discussed in
+   different working group.
+   A TEP is created to act that design work is going on on the
+   subject. The TEP is marked as `proposed`.
+6. A design is agreed after working group discussions, this new TEP
+   gets updated, and is marked as `implementable`.
+7. Work can start on the new `PipelineResource` design
+8. Once the work around this is done, the TEP gets updated to
+   `implemented` state.
+
+Later, some enhancement to the `PipelineResource` are proposed. Those
+will result in new `TEP`s.
+
+
+## Drawbacks
+
+Any additional process has the potential to engender resentment within
+the community. There is also a risk that the TEP process as designed
+will not sufficiently address the scaling challenges we face today. PR
+review bandwidth is already at a premium and we may find that the TEP
+process introduces an unreasonable bottleneck on our development
+velocity.
+
+The centrality of Git and GitHub within the TEP process also may place
```

Comment:
Ditto - do we have a strong dep to GitHub in this process? 
From a discoverability POV we could publish TEPs and their status and related artifacts on tekton website or on a dedicated website teps.tekton.dev -  setting that up could be TEP#2 :)

Reply:
We definitely should, and yes, I would see that as a TEP#2 (or another number if any TEPs is approved before that :P)